megamek/src/megamek/common/compute/Compute.java

```diff
@@ -3882,82 +3888,110 @@ public static double getAmmoAdjDamage(Game game, WeaponAttackAction weaponAttack
-              || (weaponType.getAmmoType() == AmmoTypeEnum.AC_ULTRA_THB)
-              || (weaponType.getAmmoType() == AmmoTypeEnum.AC_ROTARY)
-              || rapidAC)) {
-            return 0;
-        }
+        // Anything other than a standard AC or equivalent, UAC, or RAC does not apply
+        if (!isRapidFireAC) {
+            isRAC = weaponType instanceof RACWeapon;
+            isUAC = !isRAC && (weaponType instanceof UACWeapon);
 
-        // Get the to-hit number
-        threshold = atk.toHit(cgame).getValue();
+            if (!isRAC && !isUAC) {
+                return final_spin;
+            }
+        }
 
         // Set the weapon to single shot mode
-        weapon.setMode(rapidAC ? "" : Weapon.MODE_AC_SINGLE);
-        final_spin = 0;
+        weapon.setMode(isRapidFireAC ? "" : Weapon.MODE_AC_SINGLE);
 
-        // If weapon can't hit target, exit the function with the weapon on
-        // single shot
-        if ((threshold == TargetRoll.IMPOSSIBLE)
-              || (threshold == TargetRoll.AUTOMATIC_FAIL)) {
-            return final_spin;
-        }
-
-        // If random roll is >= to-hit + 1, then set double-spin
-        if (spinupThreshold >= threshold) {
+        // If the to-hit number is under or at the provided threshold, set multiple shots
+        if (to_hit <= spinupThreshold) {
             final_spin = 1;
-            if ((weaponType.getAmmoType() == AmmoTypeEnum.AC_ULTRA)
-                  || (weaponType.getAmmoType() == AmmoTypeEnum.AC_ULTRA_THB)) {
+            if (isUAC) {
                 weapon.setMode(Weapon.MODE_UAC_ULTRA);
-            } else if (weaponType.getAmmoType() == AmmoTypeEnum.AC_ROTARY) {
+            } else if (isRAC) {
+
                 weapon.setMode(Weapon.MODE_RAC_TWO_SHOT);
-            } else if (rapidAC) {
-                weapon.setMode(Weapon.MODE_AC_RAPID);
-            }
-        }
 
-        // If this is a Rotary cannon
-        if (weaponType.getAmmoType() == AmmoTypeEnum.AC_ROTARY) {
+                // If the to-hit number is significantly lower than the provided threshold,
+                // set for either five or six shots
 
-            // If random roll is >= to-hit + 2 then set to quad-spin
-            if (spinupThreshold >= (threshold + 1)) {
-                final_spin = 2;
-                weapon.setMode(Weapon.MODE_RAC_FOUR_SHOT);
-            }
+                if (to_hit <= (spinupThreshold - 3)) {
+                    final_spin = 5;
+                    weapon.setMode(Weapon.MODE_RAC_SIX_SHOT);
+                    return final_spin;
+                }
 
-            // If random roll is >= to-hit + 3 then set to six-spin
-            if (spinupThreshold >= (threshold + 2)) {
-                final_spin = 3;
-                weapon.setMode(Weapon.MODE_RAC_SIX_SHOT);
+                if (to_hit <= (spinupThreshold - 2)) {
+                    final_spin = 4;
+                    weapon.setMode(Weapon.MODE_RAC_FIVE_SHOT);
```

Comment:
IIRC there is a significant advantage to 3/5 shots over 4/6 due to the RAC jam roll stepping.
Or rather, the expected damage output : jamming chance is maximized for 3 and 5 shots, since they use the same jam threshold as 2/4.